test/server/http_filter_integration_test_base.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <string>
+
+#include "external/envoy/test/integration/http_integration.h"
+
+namespace Nighthawk {
+
+/**
+ * Base class with shared functionality for testing Nighthawk test server http filter extensions.
+ */
+class HttpFilterIntegrationTestBase : public Envoy::HttpIntegrationTest {
+protected:
+  /**
+   * Indicate the expected response origin.
+   */
+  enum class ResponseOrigin { UPSTREAM, EXTENSION };
+  /**
+   * Construct a new HttpFilterIntegrationTestBase instance.
+   *
+   * @param ip_version Specify the ip version that the integration test server will use to listen
+   * for connections.
+   */
+  HttpFilterIntegrationTestBase(Envoy::Network::Address::IpVersion ip_version);
+
+  /**
+   * We don't override SetUp(): tests using this fixture must call setup() instead.
+   * This is to avoid imposing the need to create a fixture per filter configuration.
+   *
+   * @param config configuration to pass to Envoy::HttpIntegrationTest::config_helper_.addFilter.
+   */
+  void initializeConfig(absl::string_view config);
+
+  /**
+   * Make getResponse send request-level configuration. Test server extensions read that
+   * configuration and merge it with their static configuration to determine a final effective
+   * configuration. See TestServerConfig in well_known_headers.h for the up to date header name.
+   *
+   * @param request_level_config Configuration to be delivered by request-header in future calls to
+   * getResponse(). For example: "{response_body_size:1024}".
+   */
+  void updateRequestLevelConfiguration(absl::string_view request_level_config);
+
+  /**
+   * Switch getResponse() to use the POST request method with an entity body.
+   * Doing so will make tests hit a different code paths in extensions.
+   */
+  void switchToPostWithEntityBody();
```

Comment:
Just a thought, curious what you think - it might be more self-documenting here, if we take in a parameter for the post body size.  Or even then, rename this to setPostBodySize, with an indication that unless this is called, the requests into the filter will be GET requests.

Reply:
I would be more up for doing this, but rather do it once a functional need for it arises (which is very possible). Right now, in the context of this PR and the changes coming after this, I'm on a mission to minimize and contain complexity that has organically grown while we added the test server and the two new extensions. Part of effort is making everything go through this hopefully well defined but minimal functionality. It should be trivial to refactor this later on according to what you suggest, once the need arises in new tests?

Reply:
That makes sense to me.  I'm happy to push it through as is, and if it comes up, do the refactor as you suggest.